test/functional/wallet_balance.py

```diff
@@ -254,8 +254,6 @@ def test_balances(*, fee_node_1=0):
         self.log.info('Put txs back into mempool of node 1 (not node 0)')
         self.nodes[0].invalidateblock(block_reorg)
         self.nodes[1].invalidateblock(block_reorg)
-        self.sync_blocks()
-        self.nodes[0].syncwithvalidationinterfacequeue()
         assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted
```

Comment:
note that wallet rpc will internally block if the chain tip changes. see `BlockUntilSyncedToCurrentChain`